### PR TITLE
termscp 0.16.0 (new formula)

### PR DESCRIPTION
**Pull request by ivaquero**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/t/termscp.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  sha256 "58f3b4770c5d1c5d7998af88b6df8c6a53dee4409f2cf6ea676caccec79cdb7f"
+  license "MIT"
+  head "https://github.com/veeso/termscp.git", branch: "main"
+
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "dbus"
+    depends_on "openssl@3"
```
Comment (chenrui333) [inline Formula/t/termscp.rb:14]: needs openssl for macos as well

File: Formula/t/termscp.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    depends_on "samba"
+  end
+
+  def install
```
Comment (chenrui333) [inline Formula/t/termscp.rb:18]:
```suggestion
  def install
    # Ensure that the `openssl` crate picks up the intended library.
    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
    ENV["OPENSSL_NO_VENDOR"] = "1"

```

File: Formula/t/termscp.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  homepage "https://termscp.veeso.dev/"
+  url "https://github.com/veeso/termscp/archive/refs/tags/v0.16.0.tar.gz"
+  sha256 "58f3b4770c5d1c5d7998af88b6df8c6a53dee4409f2cf6ea676caccec79cdb7f"
+  license "MIT"
+  head "https://github.com/veeso/termscp.git", branch: "main"
+
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "openssl@3"
+
```
Comment (chenrui333) [inline Formula/t/termscp.rb:12]:
```suggestion
  depends_on "openssl@3"

  uses_from_macos "zlib"
```

File: Formula/t/termscp.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  on_linux do
+    depends_on "dbus"
+    depends_on "samba"
+  end
+
+  def install
+    # Ensure that the `openssl` crate picks up the intended library.
+    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
+    ENV["OPENSSL_NO_VENDOR"] = "1"
+    system "cargo", "install", *std_cargo_args
```
Comment (chenrui333) [inline Formula/t/termscp.rb:22]:
```suggestion
    ENV["OPENSSL_NO_VENDOR"] = "1"

    system "cargo", "install", *std_cargo_args
```

**Review by chenrui333 — APPROVED** [review]
some minor comments.

**Comment by ivaquero** [comment]
Thank you, I will revise it later today.

**Review by chenrui333 — APPROVED** [review]

**Review by nandahkrishna — APPROVED** [review]

**Comment by chenrui333** [comment]
@ivaquero, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**